### PR TITLE
Adds the bare minimum admin components and allows admins to define list literals.

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

Adds the bare minimum admin components.
Also adds more inputs to shells.
Also adds list literals.

## Why It's Good For The Game

More tools for admins to use for events

## Changelog
:cl:
admin: Adds admin components that allow admins to set variables, get variables, call procs and execute sdql via circuits. To enable these components to function, an admin needs to enable the admin_only variable on a circuit.
qol: Adds more entity information to shell outputs to allow circuits to identify who used the shell.
expansion: Adds list literals.
/:cl:

### CODE REVIEW

**Comment by Watermelon914** [comment]
PR made with the intention to test it if any admins are willing to test it by doing an event or generally messing with it so bugs can be found and fixed.

**Comment by Time-Green** [comment]
Coding the game from inside the game???

I am both terrified and excited

**Comment by Y0SH1M4S73R** [comment]
Two concerns that would improve the flexibility of admin circuits:
- Some potential operations that could be done with admin circuits require that the return value of a proc call component be passed as the entity for another proc call component. Entity ports don't accept connections from "any" ports, so there should probably be an "as entity"-type component that adapts an "any" input into an "entity" output, if the input is something that can be represented as an entity in the first place.
- There should be some way to support nested lists in circuits, if feasible.

**Comment by coiax** [comment]
Merge only after #60222

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/modules/wiremod/components/admin/proccall.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	proc_name = add_input_port("Proc Name", PORT_TYPE_STRING)
+	arguments = add_input_port("Arguments", PORT_TYPE_LIST)
+
+	output_value = add_output_port("Output Value", PORT_TYPE_ANY)
+
+/obj/item/circuit_component/proccall/Destroy()
+	entity = null
+	proc_name = null
+	arguments = null
+	output_value = null
```
[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/wiremod/components/admin/proccall.dm:43]: ye, old code

File: code/modules/wiremod/components/list/list_literal.dm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+/obj/item/circuit_component/list_literal/save_data_to_list(list/component_data)
+	. = ..()
+	component_data["length"] = length
+
+/obj/item/circuit_component/list_literal/load_data_from_list(list/component_data)
+	set_list_size(component_data["length"])
+
+	return ..()
+
+/obj/item/circuit_component/list_literal/proc/set_list_size(new_size)
```
[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/wiremod/components/list/list_literal.dm:30]: outside of admin shenanigans, it shouldn't be possible but I can make sure.

**Pull request merged**